readthedocsext/theme/templates/projects/partials/edit/domain_list.html

```diff
@@ -38,11 +38,17 @@
      data-content="{% trans "Edit" %}">
     <i class="fa-solid fa-wrench icon"></i>
   </a>
-  <a class="ui icon button"
-     href="{% url 'projects_domains_delete' project.slug object.pk %}"
-     data-content="{% trans "Delete" %}">
-    <i class="fa-duotone fa-trash icon"></i>
-  </a>
+
+  {% url "projects_domains_delete" project_slug=project.slug domain_pk=object.pk as delete_url %}
+  {% trans "Remove domain" as action_text %}
+  {% blocktrans trimmed asvar content_text with domain=object.domain %}
+    <p>Remove domain {{ domain }}?</p>
+    <div class="ui warning message">
+      Make sure to also remove the DNS records for this domain.
+      Otherwise, another user may add the same domain to their project and serve that content from your domain (domain hijacking).
```

Comment:
This could be shortened to be easier to understand -- no need to mention users or serving content.

```suggestion
      Make sure to also remove the DNS records for this domain
      or another project might be able to reuse this domain (also known as domain hijacking).
```